interfaces/dbus/backend.go

```diff
@@ -87,6 +87,9 @@ func shouldCopyConfigFiles(snapInfo *snap.Info) bool {
 func setupDbusServiceForUserd(snapInfo *snap.Info) error {
 	coreOrSnapdRoot := snapInfo.MountDir()
 
+	// Only ever append to this list. If a file is no longer present on the
+	// root, it needs to needs to remain here so the previously-installed
+	// service file can be removed, if present.
```

Comment:
This function name/description is now a bit vague compared to what happens - I'd love to have the 'remove previously files' logic split out into it's own function which implements this cleanup logic, with this kind of description.